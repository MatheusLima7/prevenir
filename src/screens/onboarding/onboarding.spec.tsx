import { render } from "@testing-library/react-native";

import OnBoardingScreen from "./onboarding";
import { EMOTIONAL_SUPPORT } from "@src/atoms/onboarding/emotionalSupport/general.constants";

jest.mock("react-native-linear-gradient", () => jest.fn());

jest.mock("@react-native-vector-icons/FontAwesome6", () => {
  return (props: any) => `MockedIcon(${JSON.stringify(props)})`;
});

jest.mock("@react-native-vector-icons/evil-icons", () => {
  return (props: any) => `MockedIcon(${JSON.stringify(props)})`;
});

jest.mock("react-native-snap-carousel", () => {
  return {
    __esModule: true,
    default: ({ data, renderItem }: any) => {
      return data.map((item: any, index: number) =>
        renderItem({ item, index })
      );
    },
    Pagination: () => null,
  };
});

jest.mock("react-redux", () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

const mockedNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

jest.mock("@src/features/app/selectors", () => ({
  useAppTypeUserLoggedAuth: jest
    .fn()
    .mockReturnValueOnce("supporter")
    .mockReturnValueOnce("user"),
}));

describe("OnBoarding Screen Component", () => {
  it("should render component correctly", () => {
    const container = render(<OnBoardingScreen />);

    expect(container).toMatchSnapshot();
  });

  it("should render emotional supporter component for user", () => {
    const { getByTestId } = render(<OnBoardingScreen />);

    expect(getByTestId(EMOTIONAL_SUPPORT)).toBeDefined();
  });
});
