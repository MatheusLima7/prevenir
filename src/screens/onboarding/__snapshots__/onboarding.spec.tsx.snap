// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnBoarding Screen Component should render component correctly 1`] = `
<RNCSafeAreaView
  className="onboardingstyles__Wrapper-sc-agib0c-0 beDEyy"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": 160,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/heart.png",
        }
      }
    />
  </View>
  <Text
    center={true}
    color="gray"
    size="large"
    style={
      [
        {
          "color": "#717171",
          "display": "flex",
          "fontFamily": "Quicksand-SemiBold",
          "fontSize": 24,
          "fontWeight": "600",
          "lineHeight": 30,
          "textAlign": "center",
        },
        {
          "marginBottom": 0,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 0,
          "width": 280,
        },
      ]
    }
    testId="text-component"
    type="title"
  >
    Bem-vindo ao Espaço Prevenir Virtual
  </Text>
  <RCTScrollView
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    style={
      {
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "marginBottom": 24,
          }
        }
      >
        <Text
          center={true}
          color="gray"
          size="small"
          style={
            [
              {
                "color": "#717171",
                "display": "flex",
                "fontFamily": "Quicksand-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "textAlign": "center",
              },
              {
                "marginBottom": 24,
              },
            ]
          }
          testId="text-component"
          type="text"
        >
          Aqui você poderá acompanhar o seu dia a dia na superação do transtorno por uso de substâncias psicoativas, reconhecendo os seus desafios e evitando os lapsos e as recaídas.
        </Text>
      </Text>
      <Text
        style={
          {
            "marginBottom": 24,
          }
        }
      >
        <Text
          center={true}
          color="gray"
          size="small"
          style={
            [
              {
                "color": "#717171",
                "display": "flex",
                "fontFamily": "Quicksand-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "textAlign": "center",
              },
              {
                "marginBottom": 24,
              },
            ]
          }
          testId="text-component"
          type="text"
        >
          Use o aplicativo para buscar apoio nos momentos difíceis, trilhando novos caminhos para uma vida saudável. Estamos com você em cada passo dessa jornada!
        </Text>
      </Text>
      <Text
        style={
          {
            "marginBottom": 24,
          }
        }
      >
        <Text
          center={true}
          color="blue"
          size="small"
          style={
            [
              {
                "color": "#0081A7",
                "display": "flex",
                "fontFamily": "Quicksand-SemiBold",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 20,
                "textAlign": "center",
              },
              {
                "marginBottom": 24,
              },
            ]
          }
          testId="text-component"
          type="title"
        >
          Espaço Prevenir Virtual é um programa do governo do Estado de São Paulo, por meio da Secretaria de Desenvolvimento Social.
        </Text>
      </Text>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": 160,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/help.png",
        }
      }
    />
  </View>
  <Text
    center={true}
    color="gray"
    size="large"
    style={
      [
        {
          "color": "#717171",
          "display": "flex",
          "fontFamily": "Quicksand-SemiBold",
          "fontSize": 24,
          "fontWeight": "600",
          "lineHeight": 30,
          "textAlign": "center",
        },
        {
          "marginBottom": 0,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 0,
          "width": 280,
        },
      ]
    }
    testId="text-component"
    type="title"
  >
    Chamando ajuda
  </Text>
  <View
    style={
      {
        "paddingBottom": 35,
        "paddingLeft": 35,
        "paddingRight": 35,
        "paddingTop": 35,
      }
    }
  >
    <Text
      style={
        {
          "marginBottom": 24,
        }
      }
    >
      <Text
        center={true}
        color="gray"
        size="small"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="text"
      >
        Na tela principal deste aplicativo temos o botão para
      </Text>
      <Text
        center={true}
        color="blue"
        size="small"
        style={
          [
            {
              "color": "#0081A7",
              "display": "flex",
              "fontFamily": "Quicksand-SemiBold",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="title"
      >
         solicitar ajuda 
      </Text>
      <Text
        center={true}
        color="gray"
        size="small"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="text"
      >
        ao nosso espaço.
      </Text>
    </Text>
    <Text
      style={
        {
          "marginBottom": 24,
        }
      }
    >
      <Text
        center={true}
        color="gray"
        size="small"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="text"
      >
        Deixe ele
      </Text>
      <Text
        center={true}
        color="blue"
        size="small"
        style={
          [
            {
              "color": "#0081A7",
              "display": "flex",
              "fontFamily": "Quicksand-SemiBold",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="title"
      >
         pressionado 
      </Text>
      <Text
        center={true}
        color="gray"
        size="small"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="text"
      >
        para acionar o seu pedido de ajuda para nossa equipe ou para a pessoa que será seu apoiador.
      </Text>
    </Text>
    <Text
      style={
        {
          "marginBottom": 24,
        }
      }
    >
      <Text
        center={true}
        color="gray"
        size="small"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="text"
      >
        Após acionar o ajuda,
      </Text>
      <Text
        center={true}
        color="blue"
        size="small"
        style={
          [
            {
              "color": "#0081A7",
              "display": "flex",
              "fontFamily": "Quicksand-SemiBold",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="title"
      >
         entraremos em contato 
      </Text>
      <Text
        center={true}
        color="gray"
        size="small"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="text"
      >
        com você através do seu telefone.
      </Text>
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": 160,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/news.png",
        }
      }
    />
  </View>
  <Text
    center={true}
    color="gray"
    size="large"
    style={
      [
        {
          "color": "#717171",
          "display": "flex",
          "fontFamily": "Quicksand-SemiBold",
          "fontSize": 24,
          "fontWeight": "600",
          "lineHeight": 30,
          "textAlign": "center",
        },
        {
          "marginBottom": 0,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 0,
          "width": 280,
        },
      ]
    }
    testId="text-component"
    type="title"
  >
    Feed Novidades
  </Text>
  <View
    style={
      {
        "paddingBottom": 35,
        "paddingLeft": 35,
        "paddingRight": 35,
        "paddingTop": 35,
      }
    }
  >
    <Text
      style={
        {
          "marginBottom": 24,
        }
      }
    >
      <Text
        center={true}
        color="gray"
        size="small"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="text"
      >
        Acesse o
      </Text>
      <Text
        center={true}
        color="blue"
        size="small"
        style={
          [
            {
              "color": "#0081A7",
              "display": "flex",
              "fontFamily": "Quicksand-SemiBold",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="title"
      >
         FEED 
      </Text>
      <Text
        center={true}
        color="gray"
        size="small"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="text"
      >
        em nosso menu. Ali você encontrará uma série de novidades que postamos todos os dias para você.
      </Text>
    </Text>
    <Text
      style={
        {
          "marginBottom": 24,
        }
      }
    >
      <Text
        center={true}
        color="gray"
        size="small"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="text"
      >
        Desde
      </Text>
      <Text
        center={true}
        color="blue"
        size="small"
        style={
          [
            {
              "color": "#0081A7",
              "display": "flex",
              "fontFamily": "Quicksand-SemiBold",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="title"
      >
         oportunidade de trabalhos, mensagens motivacionais e informações relevantes 
      </Text>
      <Text
        center={true}
        color="gray"
        size="small"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "textAlign": "center",
            },
            {
              "marginBottom": 24,
            },
          ]
        }
        testId="text-component"
        type="text"
      >
        de cada unidade do Espaço Prevenir.
      </Text>
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItens": "center",
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 7,
        "borderTopLeftRadius": 7,
        "borderTopRightRadius": 7,
        "color": "#717171",
        "display": "flex",
        "flexDirection": "row",
        "fontSize": 14,
        "justifyContent": "space-evenly",
        "lineHeight": 20,
        "opacity": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "width": "auto",
      }
    }
  />
</RNCSafeAreaView>
`;
